server/src/main/java/org/opensearch/extensions/ExtensionRequest.java

```diff
@@ -24,41 +25,53 @@
  */
 public class ExtensionRequest extends TransportRequest {
     private static final Logger logger = LogManager.getLogger(ExtensionRequest.class);
-    private ExtensionsManager.RequestType requestType;
+    private final ExtensionsManager.RequestType requestType;
+    private final Optional<String> uniqueId;
 
     public ExtensionRequest(ExtensionsManager.RequestType requestType) {
+        this(requestType, null);
+    }
+
+    public ExtensionRequest(ExtensionsManager.RequestType requestType, Optional<String> uniqueId) {
```

Comment:
Don't make the user provide an optional.  Just take a `@Nullable String` parameter here, and then set `this.uniqueId` the same way you do in the stream constructor.  Given the overloaded constructor above, this would set the optional to null and break things later.